### DISCUSSION TITLE
jsPsych and PsiTurk

### DISCUSSION THREAD

**Discussion by kcrespo90** [jspsych/jsPsych#711] [Q&A]
Hi all,

I am student, and new to jspsych, PsiTurk and programming in general. I am using the sample-jspsych-psiturk-experiment-master go/nogo demo (2014) as a template to launch my jspsych v6.1.0 experiment in PsiTurk. 

The ad.html page and consent.html load fine. But my experiment does not launch. The plugins I used in jspsych v6.1.0 are in my PsiTurk static/js/jspsych/plugins folder, and the variables are coded the same way. Is there a reason why my jsPsych v6.1.0 plugins might not work in PsiTurk exp.html? Should I be doing something different? I would greatly appreciate some guidance!

**Comment by jodeleeuw** [comment]
I think jsPsych was at version 3 or so in the 2014 demo. There may be pieces of code that are incompatible. Do you happen to have your code on GitHub? Can you share a link?

**Reply by kcrespo90** [reply]
Thank you for your quick reply! The js, css, and lib folders can be found here: https://github.com/kcrespo90/jsPsych-PsiTurk
Is this what you need? Or do I need to upload/look somewhere different?

**Reply by jodeleeuw** [reply]
I think I also need to be able to see the exp.html.

Do you get any errors in the JS console when you try to launch the experiment?

**Reply by kcrespo90** [reply]
The exp.html page has been uploaded to my GitHub page. Before posting my question, I attempted to troubleshoot on my own and deleted several trials, so the exp.html contains a very basic code. My jsPsych version launches without errors from the jspsych-6 folder.

**Comment by jodeleeuw** [comment]
It looks like the version of jsPsych in the `js` folder is old. I'm not sure exactly which version it is, but it's older than 6.1.0 for sure. The syntax in `exp.html` also uses some of the old jsPsych syntax, specifically the `display_element` and `experiment_structure` parameters in the `jsPsych.init()` call. 

I'd suggest:

1. replacing all the code in `js` with jsPsych 6.1.0.
2. Deleting [L37](https://github.com/kcrespo90/jsPsych-PsiTurk/blob/master/exp.html#L37) and deleting [L118](https://github.com/kcrespo90/jsPsych-PsiTurk/blob/master/exp.html#L118). This will make jsPsych use the `<body>` element as the display.
3. Delete [L119](https://github.com/kcrespo90/jsPsych-PsiTurk/blob/master/exp.html#L119), which is the old way of declaring the timeline. You've got the timeline on [L116](https://github.com/kcrespo90/jsPsych-PsiTurk/blob/master/exp.html#L116) so you should be good there.
